### PR TITLE
feat: improve output structure for non-interactive modes

### LINKED ISSUE #116: Structured output improvement
**Is your feature request related to a problem? Please describe.**
We support JSON and YAML output but the structure of the output is not "developers/machine/pipe-line friendly" enough

**Describe the solution you'd like**
The output structure should be documented, clear, and consistent.

**Additional context**
YAML output: `datree test manifest.yaml -o yaml`
JSON output: `datree test manifest.yaml -o json`
Related to https://github.com/datreeio/helm-datree/issues/5#issuecomment-892128126

### PR DESCRIPTION
**Pull request by Abhra303**
This PR solves issue #116 .

The updated output structure for non-interactive options is provided by @eyarz and can be seen [here](https://www.jsonschemavalidator.net/s/vrvKDXaN). I have provided some working examples below -

Edit:
New output:

datree_structured_output

datree_json_output

### CODE REVIEW

**Comment by royhadad** [comment]
WOW! We highly appreciate your contribution to the datree project! ❤️ 

Before one of the maintainers can review your PR, we need to make sure that the CI checks are passing:

1. I have updated your branch with the latest changes from main. (no conflicts)
2. Please format your code to make the "golangci-lint" check pass. (using your IDE's auto formatter should be sufficient)
3. Also, you have a failing test, please run "make test" to view it, it causes "Travis CI - Pull Request" check to fail. (the test I see failing is: datree/cmd/test)

Let me know if you need any help, either here or via the slack channel 😊  
@Abhra303

**Comment by Abhra303** [comment]
@royhadad  thanks for the comment.

> Please format your code to make the "golangci-lint" check pass. (using your IDE's auto formatter should be sufficient)

This time I used VS Code. Don't know why it didn't format the code. Checking it now.

> Also, you have a failing test, please run "make test" to view it, it causes "Travis CI - Pull Request" check to fail. (the test I see failing is: datree/cmd/test)

I ran "make test" before committing and found various failures then. There were other failing tests that I managed to resolve but I failed to resolve this particular test failure. Because I could not figure out the meaning of the output of that test ( as I am new in testify).
Can you please help me to understand the meaning of the output?  This is the link of the output - [https://app.travis-ci.com/github/datreeio/datree/builds/244331555#L257](https://app.travis-ci.com/github/datreeio/datree/builds/244331555#L257)

File: bl/evaluation/non_interactive_evaluator.go
Comment (royhadad) [inline bl/evaluation/non_interactive_evaluator.go]: please change all the property names to lower case, example:

type NonInteractiveEvaluationResults struct {
	    formattedEvaluationResults []*FormattedEvaluationResults
	    policySummary              PolicySummary
}

(according to @eyarz draft in this comment https://github.com/datreeio/datree/issues/116#issuecomment-952767688)

File: bl/evaluation/non_interactive_evaluator.go
Comment (royhadad) [inline bl/evaluation/non_interactive_evaluator.go]: typo
RuleRresults => ruleResults

File: bl/evaluation/non_interactive_evaluator.go
Comment (royhadad) [inline bl/evaluation/non_interactive_evaluator.go]: According to https://github.com/datreeio/datree/issues/116#issuecomment-952767688
PassedPolicyCheckCount => passedPolicyValidationCount

File: bl/evaluation/non_interactive_evaluator.go
Comment (royhadad) [inline bl/evaluation/non_interactive_evaluator.go]: "RulerObject" isn't the best name, maybe "RuleResults"?

File: bl/evaluation/non_interactive_evaluator.go
Comment (royhadad) [inline bl/evaluation/non_interactive_evaluator.go]: maybe rename this file to "non_interactive_evaluator_types.go"?

File: bl/evaluation/non_interactive_evaluator.go
Comment (royhadad) [inline bl/evaluation/non_interactive_evaluator.go]: extract the "EvaluationSummary" struct into a separate type and use it in "bl/evaluation/printer.go" line 33

File: bl/evaluation/printer.go
Comment (royhadad) [inline bl/evaluation/printer.go]: If this happens (and it shouldn't), shouldn't we throw an error? this will just silence the error.

File: cmd/test/main.go
Comment (royhadad) [inline cmd/test/main.go]: we can remove this check and just return this:
`return validationManager, createEvaluationResponse, results, err`

it is logically the same 😊

File: bl/evaluation/evaluator.go
Comment (royhadad) [inline bl/evaluation/evaluator.go]: remove "rulesCount" and "policyName" from the arguments list
replace evaluationId with cliClient.CreateEvaluationResponse, it includes all 3 of the properties

Comment (royhadad) [inline bl/evaluation/evaluator.go]: I suggest replacing "nonInteractiveMode" with "isInteractiveMode", more readable, avoids the double negative

File: bl/evaluation/evaluator.go
Comment (royhadad) [inline bl/evaluation/evaluator.go]: `ruleRObject` looks like a typo

File: bl/evaluation/printer_test.go
Comment (royhadad) [inline bl/evaluation/printer_test.go]: you can reuse the struct from non_interactive_evaluator.go

File: cmd/test/main_test.go
Comment (royhadad) [inline cmd/test/main_test.go]: This is not ideal, although I couldn't fix it myself.
If you can't fix this (as well as the other 3 occurrences, that is OK, I will try to fix it later)

**Review by royhadad — CHANGES_REQUESTED** [review]

**Comment by royhadad** [comment]
@Abhra303 
GREAT JOB! overall - looks good!
I added some changes requests, once you fix those we can merge this PR 😊

File: cmd/test/main_test.go
Comment (Abhra303) [reply cmd/test/main_test.go]: yeah, you're right. I tried my best to fix it in the ideal way. But for the lack of my knowledge in testify I couldn't fix it. 
Although, I will try to fix it now.

File: bl/evaluation/non_interactive_evaluator.go
Comment (Abhra303) [reply bl/evaluation/non_interactive_evaluator.go]: okay :)

File: bl/evaluation/non_interactive_evaluator.go
Comment (Abhra303) [reply bl/evaluation/non_interactive_evaluator.go]: Actually, in the [draft](https://www.jsonschemavalidator.net/s/vrvKDXaN) structure, there is `ruleRresults` . that's why I defined `RuleRresults` here. Should I rename it to `ruleResults` then?

File: bl/evaluation/non_interactive_evaluator.go
Comment (Abhra303) [reply bl/evaluation/non_interactive_evaluator.go]: okay. fixing it.

File: bl/evaluation/non_interactive_evaluator.go
Comment (Abhra303) [reply bl/evaluation/non_interactive_evaluator.go]: I am unskilled when it comes to naming😅
changing it.

File: bl/evaluation/non_interactive_evaluator.go
Comment (Abhra303) [reply bl/evaluation/non_interactive_evaluator.go]: okay.

File: bl/evaluation/non_interactive_evaluator.go
Comment (Abhra303) [reply bl/evaluation/non_interactive_evaluator.go]: what would be the name of that type?

File: bl/evaluation/printer.go
Comment (Abhra303) [reply bl/evaluation/printer.go]: okay.

File: bl/evaluation/non_interactive_evaluator.go
Comment (royhadad) [reply bl/evaluation/non_interactive_evaluator.go]: yep! it was a typo

File: bl/evaluation/non_interactive_evaluator.go
Comment (royhadad) [reply bl/evaluation/non_interactive_evaluator.go]: type `EvaluationSummary` sounds good

File: bl/evaluation/non_interactive_evaluator.go
Comment (Abhra303) [reply bl/evaluation/non_interactive_evaluator.go]: I didn't change the property names in lower case because `yaml.Marshal`, `json.Marshal` and `xml.Marshal` do not support field names that start with lower cases. See [this](https://pkg.go.dev/gopkg.in/yaml.v2#Marshal).

Instead I used struct tag to do this.

File: bl/evaluation/non_interactive_evaluator.go
Comment (Abhra303) [reply bl/evaluation/non_interactive_evaluator.go]: I changed the name to `RuleResult` here - as `RuleResults` is an array of `RuleResult`.

File: bl/evaluation/non_interactive_evaluator.go
Comment (Abhra303) [reply bl/evaluation/non_interactive_evaluator.go]: done

**Comment by Abhra303** [comment]
@royhadad could you please review the last change?
Thanks :)

**Review by royhadad — APPROVED** [review]

**Comment by royhadad** [comment]
Merged, thank you for the contribution 😊

**Pull request merged**
